File: ClientApp/src/app/views/configurarion/ConfigurationModel.ts
export class RequestModel {
  ReqID: any;
}
export class urldropdown {
  text: any;
  value: any;
}

export class ConfigureModel {
  Account_id: any;
  VMN: any;
  VMN_length_flag: any;
  ReqID: any;
  UrlID: any;
  url: any;
  method: any;
  date_format: any;
  content_type: any; 
  proxy: any;
  header: any;
  token: any;
  basic: any;
  //vmn: any;
  header_keys = Array<headerkeys>();
  parameters_keys = Array<parameterkeys>(); 
  success_response: any;
  failure_response: any; 
  timeout: any;

  retry_count: any; authentication: any;
  //VMN_length_flag: any;
  actual_vmn_list:  Array<vmnlist>;

}

export class UrlTestModel {
  ReqID: any;
  UrlID: any;
  url_b: any;
  method_b: any;
  date_format_b: any;
  content_type_b: any;
  proxy_b: any;
  header_b: any;
  authentication_b: any;
  token_b: any;
  basic_b: any;
  header_keys_b = Array<UrlTestHeaderkeys>();
  parameters_keys_b = Array<UrlTestParameterkeys>();
  response_param_b: any;
  success_b: any;
  failure_b: any;
  timeout_b: any;
  retry_count_b: any;
  data: any;
  parameters: any;
}

export class headerkeys {
  key: any;
  value: any;
}

export class parameterkeys {
  parameter1: any;
  value1: any;
  parameter2: any;
  value2: any;
  parameter3: any;
  value3: any;
  parameter4: any;
  value4: any;
  parameter5: any;
  value5: any;
  parameter6: any;
  value6: any;
  parameter7: any;
  value7: any;
  parameter8: any;
  value8: any;
  parameter9: any;
  value9: any;
  parameter10: any;
  value10: any;
}

export class UrlTestRespParameter {
  //parameter1: any;
  //value1: any;
  //parameter2: any;
  //value2: any;
  //parameter3: any;
  //value3: any;
  //parameter4: any;
  //value4: any;
  //parameter5: any;
  //value5: any;
  //parameter6: any;
  //value6: any;
  //parameter7: any;
  //value7: any;
  //parameter8: any;
  //value8: any;
  //parameter9: any;
  //value9: any;
  //parameter10: any;
  //value10: any;
  key: any;
  value: any;
}

export class UrlTestHeaderkeys {
  key: any;
  value: any;
}

export class UrlTestParameterkeys {
  parameter1: any;
  value1: any;
  parameter2: any;
  value2: any;
  parameter3: any;
  value3: any;
  parameter4: any;
  value4: any;
  parameter5: any;
  value5: any;
  parameter6: any;
  value6: any;
  parameter7: any;
  value7: any;
  parameter8: any;
  value8: any;
  parameter9: any;
  value9: any;
  parameter10: any;
  value10: any;
}

export class adduser {
  Account_id: any;
  ReqID: any;
  CustomerName: any;
  UserName: any;
  MobileNumber: any;
  EmailId: any;
  Address: any;
}

export class NewRequestModel {
  account_id: any;
  customername: any;
  circle: any;
  plan: any;
  service: any;
  platform: any;
}

export class AddVmn {
  Request_id: any;
  account_id: any;
  vmnlist: Array<vmnlist>;
}
export class vmnlist {
  msisdn: any;
  imsi: any;
}

File: Models/ConfigurationModel.cs
namespace SMS.Models
{
    public class ConfigurationModel
    {
        public string? ReqID { get; set; }
    }
    public class InserturltestModel { 
    public string? data { get; set; }
        public string? url_id { get; set; }
        public string? parameters { get; set; }
    }
    public class RequestModel 
    { 
        public string? data { get; set; }
    }

    public class ConfigureModel
    {
        //public string? proxy_ { get; set; }
        //public string? authentication_ { get; set; }
        public string? VMN { get; set; }
        public string? Account_id { get; set; }
        public string? ReqID { get; set; }
        public string? url { get; set; }
        public string? method { get; set; }
        public string? date_format { get; set; }
        public string? content_type { get; set; }
        public string? proxy{ get; set; }
        public string? splitter { get; set; }
        public List<HeaderKey>? header_keys { get; set; } // Updated to List<HeaderKey>
        public List<ParameterKey>? parameters_keys { get; set; } // Updated to List<ParameterKey>
        public string? authentication { get; set; }
        public string? token { get; set; }
        public string? basic { get; set; }
        public string? success_response { get; set; }
        public string? failure_response { get; set; }
        public string? retry_count { get; set; }
        public string? timeout { get; set; }
        public string? header { get; set; }

        public string? Shortcode_flag { get; set; }
        public string? Unicode_flag { get; set; }
        public string? VMN_length_flag { get; set; }
        public List<Vmnlist>? actual_vmn_list { get; set; }

        //public vmnlist { get; set; }
    }

    public class UrlTestModel
    {
        public string? ReqID { get; set; }
        public string? UrlID { get; set; }
        public string? url_b { get; set; }
        public string? method_b { get; set; }
        public string? date_format_b { get; set; }
        public string? content_type_b { get; set; }
        public string? proxy_b { get; set; }
        public string? header_b { get; set; }
        public List<UrlTestHeaderKey>? header_keys_b { get; set; } // Updated to List<HeaderKey>
        public List<UrlTestParameterKey>? parameters_keys_b { get; set; } // Updated to List<ParameterKey>
        public string? authentication_b { get; set; }
        public string? token_b { get; set; }
        public string? basic_b { get; set; }
        public string? success_b { get; set; }
        public string? failure_b { get; set; }
        public string? retry_count_b { get; set; }
        public string? timeout_b { get; set; }
        public List<UrlTestRespParameter>? response_param_b { get; set; }
        public string? response3 { get; set; }
        public string? parameters { get; set; }
        public string? data { get; set; }
    }

    public class HeaderKey
    {
        public string? key { get; set; }
        public string? value { get; set; }
    }

    public class ParameterKey
    {
        public string? parameter1 { get; set; }
        public string? value1 { get; set; }
        public string? parameter2 { get; set; }
        public string? value2 { get; set; }
        public string? parameter3 { get; set; }
        public string? value3 { get; set; }
        public string? parameter4 { get; set; }
        public string? value4 { get; set; }
        public string? parameter5 { get; set; }
        public string? value5 { get; set; }
        public string? parameter6 { get; set; }
        public string? value6 { get; set; }
        public string? parameter7 { get; set; }
        public string? value7 { get; set; }
        public string? parameter8 { get; set; }
        public string? value8 { get; set; }
        public string? parameter9 { get; set; }
        public string? value9 { get; set; }
        public string? parameter10 { get; set; }
        public string? value10 { get; set; }
    }

    public class UrlTestHeaderKey
    {
        public string? key { get; set; }
        public string? value { get; set; }
    }

    public class UrlTestParameterKey
    {
        public string? parameter1 { get; set; }
        public string? value1 { get; set; }
        public string? parameter2 { get; set; }
        public string? value2 { get; set; }
        public string? parameter3 { get; set; }
        public string? value3 { get; set; }
        public string? parameter4 { get; set; }
        public string? value4 { get; set; }
        public string? parameter5 { get; set; }
        public string? value5 { get; set; }
        public string? parameter6 { get; set; }
        public string? value6 { get; set; }
        public string? parameter7 { get; set; }
        public string? value7 { get; set; }
        public string? parameter8 { get; set; }
        public string? value8 { get; set; }
        public string? parameter9 { get; set; }
        public string? value9 { get; set; }
        public string? parameter10 { get; set; }
        public string? value10 { get; set; }
    }

    public class UrlTestRespParameter
    {
        public string? key { get; set; }
        public string? value { get; set; }
    }

    public class Adduser {
        public string? ReqID { get; set; }
        public string? Account_id { get; set; }
        public string? CustomerName { get; set; }
        public string? UserName { get; set; }
        public string? MobileNumber { get; set; }
        public string? EmailId { get; set;}
        public string? Address { get; set; }

    }

    public class CreateNewRequestModel
    {
        public string? account_id { get; set; }
        public string? customername { get; set; }
        public string? circle { get; set; }
        public string? plan { get; set; }
        public string? service { get; set; }
        public string? platform { get; set; }
    }
  
    public class AddVmnModel 
    { 
        public string? Request_id { get; set; }
        public string? account_id { get; set; }
        public List<Vmnlist>? vmnlist { get; set; }
    }

    public class Vmnlist 
    { 
        public string? msisdn { get; set; }
        public string? imsi { get; set; }
    }
}
